app/src/main/java/net/gsantner/markor/frontend/textview/LineNumbersTextView.java

```diff
@@ -0,0 +1,302 @@
+package net.gsantner.markor.frontend.textview;
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.Rect;
+import android.text.Editable;
+import android.text.Layout;
+import android.text.TextWatcher;
+import android.util.AttributeSet;
+import android.widget.EditText;
+
+import androidx.appcompat.widget.AppCompatTextView;
+
+/**
+ * @author Li Guanglin
+ */
+public class LineNumbersTextView extends AppCompatTextView {
```

Comment:
We disable some accessibility things in Highlighting editor for performance. May be good to disable them here too